### PR TITLE
Add regrid to sispeed derivation when siu and siv are not in the same grid

### PR DESCRIPTION
**Pull request by jvegasbsc**
Adding regrid to sispeed derivation

-   [ ] [Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do, if you haven't done so already (and add the link at the bottom)
-   [x] This pull request has a descriptive title that can be used in a changelog
-   [x] Add unit tests
-   [x] Public functions should have a numpy-style docstring so they appear properly in the [API documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/api/esmvalcore.html). For all other functions a one line docstring is sufficient.
-   [x] If writing a new/modified preprocessor function, please update the [documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/esmvalcore/preprocessor.html)
-   [x] Circle/CI tests pass. Status can be seen below your pull request. If the tests are failing, click the link to find out why.
-   [x] Codacy code quality checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why. If you suspect Codacy may be wrong, please ask by commenting.
-   [x] Please use `yamllint` to check that your YAML files do not contain mistakes
-   [x] If you make backward incompatible changes to the recipe format, make a new pull request in the [ESMValTool repository](https://github.com/ESMValGroup/ESMValTool) and add the link below

### CODE REVIEW

File: esmvalcore/preprocessor/_derive/sispeed.py [hunk 1]
```diff
@@ -39,43 +39,13 @@ def calculate(cubes):
...
-                                  siu.coord_dims('latitude'))
-                siv.add_dim_coord(siu.coord('longitude'),
-                                  siu.coord_dims('longitude'))
-            else:
-                siv.add_aux_coord(siu.coord('latitude'),
-                                  siu.coord_dims('latitude'))
-                siv.add_aux_coord(siu.coord('longitude'),
-                                  siu.coord_dims('longitude'))
+            logger.debug('Regridding siv into siu grid to compute sispeed')
+            siv = regrid(siv, siu, 'linear')
```
Comment (valeriupredoi) [inline esmvalcore/preprocessor/_derive/sispeed.py]: is `siu` the right grid? why not regrid `siu` on `siv` grid? Or it doesn't actually matter at all

Comment (jvegasbsc) [reply esmvalcore/preprocessor/_derive/sispeed.py]: It does not matter. In NEMO, for example, siu and siv are computed on the u and v points of the cell:

image

Any of those will be correct (in our case, probably even more correct will be to interpolate to the T point, which is usually the one used for arecello and other grid points)

Comment (valeriupredoi) [reply esmvalcore/preprocessor/_derive/sispeed.py]: love it - a diagram!

**Comment by jvegasbsc** [comment]
Required for ESMValGroup/ESMValTool#1079

**Pull request merged**